### PR TITLE
Fix user locale is set correctly test in UserServiceIntTest

### LINKED ISSUE #8313: UserServiceIntTest fails in microservice with OAuth 2.0
##### **Overview of the issue**

I created a microservice with OAuth 2.0 and MongoDB and `UserServiceIntTest.assertThatUserLocaleIsCorrectlySetFromAuthenticationDetails()` fails when I run `mvn test`.

```
[INFO] Scanning for projects...
[INFO] 
[INFO] -------------------< com.okta.developer.store:store >-------------------
[INFO] Building Store 0.0.1-SNAPSHOT
[INFO] --------------------------------[ war ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:copy-resources (default-resources) @ store ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 6 resources
[INFO] Copying 7 resources
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ store ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 6 resources
[INFO] Copying 7 resources
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-versions) @ store ---
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:copy-resources (docker-resources) @ store ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 6 resources
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:prepare-agent (pre-unit-tests) @ store ---
[INFO] argLine set to -javaagent:/Users/mraible/.m2/repository/org/jacoco/org.jacoco.agent/0.8.2/org.jacoco.agent-0.8.2-runtime.jar=destfile=/Users/mraible/apps/store/target/test-results/coverage/jacoco/jacoco.exec -Djava.security.egd=file:/dev/./urandom -Xmx256m
[INFO] 
[INFO] --- maven-compiler-plugin:3.7.0:compile (default-compile) @ store ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 79 source files to /Users/mraible/apps/store/target/classes
[INFO] /Users/mraible/apps/store/src/main/java/com/okta/developer/store/security/oauth2/SimpleAuthoritiesExtractor.java: /Users/mraible/apps/store/src/main/java/com/okta/developer/store/security/oauth2/SimpleAuthoritiesExtractor.java uses unchecked or unsafe operations.
[INFO] /Users/mraible/apps/store/src/main/java/com/okta/developer/store/security/oauth2/SimpleAuthoritiesExtractor.java: Recompile with -Xlint:unchecked for details.
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ store ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.7.0:testCompile (default-testCompile) @ store ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 15 source files to /Users/mraible/apps/store/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ store ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.okta.developer.store.service.UserServiceIntTest

        ██╗ ██╗   ██╗ ████████╗ ███████╗   ██████╗ ████████╗ ████████╗ ███████╗
        ██║ ██║   ██║ ╚══██╔══╝ ██╔═══██╗ ██╔════╝ ╚══██╔══╝ ██╔═════╝ ██╔═══██╗
        ██║ ████████║    ██║    ███████╔╝ ╚█████╗     ██║    ██████╗   ███████╔╝
  ██╗   ██║ ██╔═══██║    ██║    ██╔════╝   ╚═══██╗    ██║    ██╔═══╝   ██╔══██║
  ╚██████╔╝ ██║   ██║ ████████╗ ██║       ██████╔╝    ██║    ████████╗ ██║  ╚██╗
   ╚═════╝  ╚═╝   ╚═╝ ╚═══════╝ ╚═╝       ╚═════╝     ╚═╝    ╚═══════╝ ╚═╝   ╚═╝

:: JHipster 🤓  :: Running Spring Boot 2.0.4.RELEASE ::
:: https://www.jhipster.tech ::

2018-09-13 12:00:34.130  INFO 86449 --- [           main] c.o.d.store.config.MetricsConfiguration  : Initializing Metrics Log reporting
2018-09-13 12:00:37.316  WARN 86449 --- [           main] c.o.d.store.config.CacheConfiguration    : No discovery service is set up, Hazelcast cannot create a cluster.
2018-09-13 12:00:38.771  WARN 86449 --- [           main] com.hazelcast.instance.Node              : [10.130.0.63]:5701 [dev] [3.9.4] No join method is enabled! Starting standalone.
2018-09-13 12:00:40.457  WARN 86449 --- [           main] c.n.c.sources.URLConfigurationSource     : No URLs will be polled as dynamic configuration sources.
2018-09-13 12:00:40.467  WARN 86449 --- [           main] c.n.c.sources.URLConfigurationSource     : No URLs will be polled as dynamic configuration sources.
2018-09-13 12:00:49.488  INFO 86449 --- [           main] c.o.d.store.config.LoggingConfiguration  : Initializing Logstash logging
2018-09-13 12:00:49.561  INFO 86449 --- [           main] c.o.d.store.config.LoggingConfiguration  : Filtering metrics logs from all appenders except the LOGSTASH appender
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 21.249 s <<< FAILURE! - in com.okta.developer.store.service.UserServiceIntTest
[ERROR] assertThatUserLocaleIsCorrectlySetFromAuthenticationDetails(com.okta.developer.store.service.UserServiceIntTest)  Time elapsed: 0.1 s  <<< ERROR!
org.springframework.dao.IncorrectResultSizeDataAccessException: Query { "$java" : Query: { "login" : "anonymoususer" }, Fields: { }, Sort: { } } returned non unique result.
	at com.okta.developer.store.service.UserServiceIntTest.assertThatUserLocaleIsCorrectlySetFromAuthenticationDetails(UserServiceIntTest.java:116)

2018-09-13 12:00:50.808  INFO 86449 --- [      Thread-14] c.o.d.store.config.CacheConfiguration    : Closing Cache Manager
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   UserServiceIntTest.assertThatUserLocaleIsCorrectlySetFromAuthenticationDetails:116 » IncorrectResultSizeDataAccess
[INFO] 
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 28.193 s
[INFO] Finished at: 2018-09-13T12:00:51+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project store: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/mraible/apps/store/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

```

##### **Motivation for or Use Case**

All tests should pass after creating a new app.

##### **Reproduce the error**

Create a microservices app with OAuth 2.0, MongoDB, and Elasticsearch, then run `mvn test`.

##### **JHipster Version(s)**

master branch

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.okta.developer.store",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.3.1",
    "applicationType": "microservice",
    "baseName": "store",
    "packageName": "com.okta.developer.store",
    "packageFolder": "com/okta/developer/store",
    "serverPort": "8082",
    "authenticationType": "oauth2",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "mongodb",
    "devDatabaseType": "mongodb",
    "prodDatabaseType": "mongodb",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "enableTranslation": true,
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "clientPackageManager": "npm",
    "nativeLanguage": "en",
    "languages": [
      "en"
    ],
    "skipClient": true,
    "skipUserManagement": true
  }
}
</pre>
</details>

NOTE: the rest of `jhipster info` fails with the following error message:

```
##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

The passed validation must be valid to be added to the field.
Errors: No value
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: ERROR!
Error while parsing entities to JDL

    at Environment.error (/Users/mraible/dev/generator-jhipster/node_modules/yeoman-environment/lib/environment.js:157:40)
    at module.exports.error (/Users/mraible/dev/generator-jhipster/generators/generator-base.js:2058:18)
    at module.exports.generateJDLFromEntities (/Users/mraible/dev/generator-jhipster/generators/generator-base-private.js:1003:18)
    at module.exports.displayEntities (/Users/mraible/dev/generator-jhipster/generators/info/index.js:57:34)
    at Object.<anonymous> (/Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/lib/index.js:418:27)
    at /Users/mraible/dev/generator-jhipster/node_modules/run-async/index.js:25:25
    at new Promise (<anonymous>)
    at /Users/mraible/dev/generator-jhipster/node_modules/run-async/index.js:24:19
    at self.env.runLoop.add.completed (/Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/lib/index.js:419:13)
    at runCallback (timers.js:810:20)
Execution time: 3 s.
```

My `.jhipster` directory contains a `Product.json`:

```json
{
    "fluentMethods": true,
    "clientRootFolder": "store",
    "relationships": [],
    "fields": [
        {
            "fieldName": "name",
            "fieldType": "String",
            "fieldValidateRules": [
                "required"
            ]
        },
        {
            "fieldName": "price",
            "fieldType": "Double",
            "fieldValidateRules": [
                "required",
                "min"
            ],
            "fieldValidateRulesMin": 0
        },
        {
            "fieldName": "image",
            "fieldType": "byte[]",
            "fieldTypeBlobContent": "image"
        }
    ],
    "dto": "no",
    "searchEngine": "elasticsearch",
    "service": "no",
    "entityTableName": "product",
    "jpaMetamodelFiltering": false,
    "pagination": "pagination",
    "microserviceName": "store"
}
```

### PR DESCRIPTION
**Pull request by mraible**
Fix #8313.

### CODE REVIEW

**Comment by DanielFran** [comment]
Restarted failing tests

**Comment by mraible** [comment]
It seems my fix doesn't work:

```
[INFO] Running io.github.jhipster.travis.service.UserServiceIntTest
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.357 s <<< FAILURE! - in io.github.jhipster.travis.service.UserServiceIntTest
[ERROR] assertThatUserLocaleIsCorrectlySetFromAuthenticationDetails(io.github.jhipster.travis.service.UserServiceIntTest)  Time elapsed: 0.141 s  <<< ERROR!
org.springframework.orm.jpa.JpaSystemException: ids for this class must be manually assigned before calling save(): io.github.jhipster.travis.domain.User; nested exception is org.hibernate.id.IdentifierGenerationException: ids for this class must be manually assigned before calling save(): io.github.jhipster.travis.domain.User
	at io.github.jhipster.travis.service.UserServiceIntTest.assertThatUserLocaleIsCorrectlySetFromAuthenticationDetails(UserServiceIntTest.java:110)
Caused by: org.hibernate.id.IdentifierGenerationException: ids for this class must be manually assigned before calling save(): io.github.jhipster.travis.domain.User
	at io.github.jhipster.travis.service.UserServiceIntTest.assertThatUserLocaleIsCorrectlySetFromAuthenticationDetails(UserServiceIntTest.java:110)
```

File: generators/server/templates/src/test/java/package/service/UserServiceIntTest.java.ejs [hunk 1]
```diff
@@ -400,12 +400,6 @@ public class UserServiceIntTest <% if (databaseType === 'cassandra') { %>extends
     @Transactional
     <%_ } _%>
     public void assertThatUserLocaleIsCorrectlySetFromAuthenticationDetails() {
-        user.setId(Constants.ANONYMOUS_USER);
-        user.setLogin(Constants.ANONYMOUS_USER);
```
Comment (ruddell) [inline generators/server/templates/src/test/java/package/service/UserServiceIntTest.java.ejs]: Leaving lines 403 and 404 seems to work based on a test I did ([Travis Results](https://travis-ci.org/ruddell/generator-jhipster/builds/431242491), [Commit](https://github.com/ruddell/generator-jhipster/commit/f3ae6d8a7ef9))

**Comment by deepu105** [comment]
@mraible does it work now, can we merge it?

**Comment by pascalgrimaud** [comment]
Not sure it is fixed correctly.
I still see the build failing:
https://travis-ci.org/hipster-labs/jhipster-travis-build/jobs/432385324

With the log:
```

[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   UserServiceIntTest.assertThatUserLocaleIsCorrectlySetFromAuthenticationDetails:113 » JpaSystem
[INFO] 
[ERROR] Tests run: 388, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:58 min
[INFO] Finished at: 2018-09-24T19:34:20+10:00
[INFO] ------------------------------------------------------------------------
```

**Comment by ruddell** [comment]
It looks like this fix works for maven but breaks gradle (adding the if statement back to the code fixes it for gradle, but breaks maven)

**Comment by sendilkumarn** [comment]
I tested it locally before merging and it was working fine for Maven (default)

**Pull request merged**